src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -2628,6 +2622,7 @@ private void CheckSubmissionCompletenessAndRemove(BuildSubmission submission)
                 // If the submission has completed or never started, remove it.
                 if (submission.IsCompleted || submission.BuildRequest == null)
                 {
+                    _overallBuildSuccess &= (submission.BuildResult?.OverallResult == BuildResultCode.Success);
```

Comment:
The `submission.BuildRequest == null` seems to be a difference from previous behavior - when this can happen? And is that expected to be a build failure indicator?

Reply:
Normally one call `submission = Build.ManagerPendBuildReuest(); ... ; submission.Execute();` If for any reason code check submission for completness even though it has not yet started, cancel scenarios mostly I guess, submission is removed from list of submissions. 
I believe report failure of overall build manager i n such scenarios looks reasonable to me.
If we, however, receive reports from field, we can fix it easily by not changing _overallBuildSuccess if build submission has not started yet.